src/sentry/integrations/vercel/webhook.py

```diff
@@ -142,27 +143,53 @@ def post(self, request: Request) -> Response:
 
         # Vercel's webhook allows you to subscribe to different events,
         # denoted by the `type` attribute. We currently subscribe to:
-        #     * integration-configuration-removed (Configuration Removed)
-        #     * deployment (Deployment Created)
+        #     * integration-configuration-removed, integration-configuration.removed (Configuration Removed)
+        #     * deployment, deployment.created (Deployment Created)
         # https://vercel.com/docs/integrations/webhooks-overview
-        try:
-            event_type = request.data["type"]
-        except KeyError:
-            return self.respond({"detail": "Missing event type."}, status=400)
+        with configure_scope() as scope:
+            try:
+                event_type = request.data["type"]
+            except KeyError:
+                return self.respond({"detail": "Missing event type."}, status=400)
+
+            # Try the new Vercel request. If it fails, try the old Vercel request
+            try:
+                payload = request.data["payload"]
+                external_id = (
+                    payload.get("team")["id"] if payload.get("team") else payload["user"]["id"]
+                )
+                scope.set_tag("vercel_webhook.type", "new")
 
-        external_id = request.data.get("teamId") or request.data["userId"]
+                if event_type == "integration-configuration.removed":
+                    configuration_id = request.data["payload"]["configuration"]["id"]
```

Comment:
nit: You already have `payload` in scope, so you can just do `payload["configuration"]["id"]`

Reply:
Good catch, thank you!